### DISCUSSION TITLE
Customizing configuration page for a user federation provider

### DISCUSSION THREAD

**Discussion by eitchugo** [keycloak/keycloak#27332] [Archive]
Hi folks, I developed a user federation custom provider and it's working just fine. I want to tweak the provider settings page and I'm not finding how to do it... For example: I want to add section headings like `Connecting Settings`, `Database settings`, `Synchronization Settings`, etc. Another example: I want to give some fields a `required` asterisk (*), instead of every field not having one. I can see working examples on built-in configuration pages for LDAP, but couldn't find anything for custom providers.

I found some react files on source (`js/apps/admin-ui/src/user-federation`) including which seems the ldap and custom files, but couldn't use/replace them on my jar. I also searched a lot of examples and other providers but couldn't find anyone that customizes the configuration page.. Any advice on that? :)

Using latest version (`23.0.7`)

configuration page screen
![Screenshot_20240227_221048](https://github.com/keycloak/keycloak/assets/349457/3eeba3f6-d4e6-43d3-ba5c-9f404ece8802)

some elements I wanted from the ldap configuration page
![Screenshot_20240227_221154-1](https://github.com/keycloak/keycloak/assets/349457/d891ae09-1f60-4180-a1c0-055cdf7695de)

**Comment by kesavkolla** [comment]
I am also looking for customizing the federation provider. Is there anyway to have custom screen deployed?
